visualization/app/codeCharta/ui/legendPanel/legendPanel.component.ts

```diff
@@ -53,9 +87,30 @@ export class LegendPanelController
 
 	onColorMetricChanged(colorMetric: string) {
 		this._viewModel.colorMetric = colorMetric
+		this._viewModel.colorMetricDescription = getMetricDescriptions().get(this._viewModel.colorMetric)
 		this.updateMaxMetricValue()
 	}
 
+	onHeightMetricChanged(heightMetric: string) {
+		this._viewModel.heightMetric = heightMetric
+		this._viewModel.heightMetricDescription = getMetricDescriptions().get(this._viewModel.heightMetric)
+	}
+
+	onAreaMetricChanged(areaMetric: string) {
+		this._viewModel.areaMetric = areaMetric
+		this._viewModel.areaMetricDescription = getMetricDescriptions().get(this._viewModel.areaMetric)
+	}
+
+	onEdgeMetricChanged(edgeMetric: string) {
+		this._viewModel.edge = edgeMetric
+		this._viewModel.edgeMetricData = this.storeService.getState().metricData.edgeMetricData
```

Comment:
I cannot see that edgeMetricData is used in the template. Therefore, we should not pass it to the view model.